modules/core/map/js/farm_map.js

```diff
@@ -79,6 +34,69 @@
           }
         });
       }
+
+    },
+
+    createMapInstance: function(context, element, drupalSettingsKey, mapInstanceOptions) {
+
+      // Get the units.
+      let units = 'metric';
+      if (!!drupalSettings.farm_map.units) {
+        units = drupalSettings.farm_map.units;
+      }
+
+      // Build default options.
+      const defaultOptions = {
+        units,
+        interactions: {
+          onFocusOnly: true
+        },
+      };
+
+      const mapOptions = {
+        ...defaultOptions,
+        ...drupalSettings.farm_map[drupalSettingsKey].instance,
+        ...mapInstanceOptions
+      };
+      const instance = farmOS.map.create(element, mapOptions);
+      instance.drupalSettingsKey = drupalSettingsKey;
```

Comment:
Ah interesting. This could be a solution! Could it be abstracted a bit so that `instance.settings()` was a method provided by the `createMapInstance` factory method? That way a behavior wouldn't have to navigate the `drupalSettings` array.

`instance.settings = () => drupalSettings.farm_map[drupalSettingsKey].instance;`

This could maybe be a staging area before something more standardized was adopted in farmOS-map core